contracts/discrepancies/nonce/InternalCallee.sol

```diff
@@ -1,28 +1,50 @@
 // SPDX-License-Identifier: Apache-2.0
 pragma solidity ^0.8.0;
 
-contract Sample {}
+contract Sample {
+  function selfdestruct() external {
+    selfdestruct(payable(msg.sender));
+  }
+}
 
 contract InternalCallee {
-    uint calledTimes = 0;
+  uint calledTimes = 0;
+
+  function factorySample() external returns (address) {
+    return address(new Sample());
+  }
+
+  function externalFunction() external returns (uint) {
+    return ++calledTimes;
+  }
+
+  function revertWithRevertReason() public returns (bool) {
+    revert('RevertReason');
+  }
+
+  function revertWithoutRevertReason() public returns (bool) {
+    revert();
+  }
+
+  function selfdestruct(address payable _addr) external {
```

Comment:
this will throw
```bash
Warning: This declaration shadows a builtin symbol.
```

Change it like below to avoid the warning
```suggestion
  function selfDestruct() external {
```

Reply:
renamed to selfDestructSample because we already have selfDestruct function in the same file